packages/flutter/lib/src/foundation/collections.dart

```diff
@@ -4,16 +4,16 @@
 
 // TODO(ianh): These should be on the Set and List classes themselves.
 
-/// Compares two sets for deep equality.
+/// Compares two sets for shallow equality.
 ///
 /// Returns true if the sets are both null, or if they are both non-null, have
 /// the same length, and contain the same members. Returns false otherwise.
 /// Order is not compared.
 ///
-/// The term "deep" above refers to the first level of equality: if the elements
-/// are maps, lists, sets, or other collections/composite objects, then the
-/// values of those elements are not compared element by element unless their
-/// equality operators ([Object.==]) do so.
+/// If the elements in the sets are collections (e.g., maps, lists, sets),
```

Comment:
I don't think these changes are needed if you use "element-by-element equality" above, and they seem to make it less clear to me.

Adding a "consider" note for `DeepCollectionEquality` is still a good idea, though.

Reply:
@Piinks has also pointed out the problem with this sentence, and I agree with you.